packages/integrations/vercel/src/index.ts

```diff
@@ -0,0 +1,79 @@
+import type { AstroIntegration, AstroConfig } from 'astro';
+import type { IncomingMessage, ServerResponse } from 'http';
+import type { PathLike } from 'fs';
+
+import fs from 'fs/promises';
+import { fileURLToPath } from 'url';
+import { globby } from 'globby';
+import esbuild from 'esbuild';
+
+export type VercelRequest = IncomingMessage;
+export type VercelResponse = ServerResponse;
+export type VercelHandler = (request: VercelRequest, response: VercelResponse) => void | Promise<void>;
+
+const writeJson = (path: PathLike, data: any) => fs.writeFile(path, JSON.stringify(data), { encoding: 'utf-8' });
+
+const ENDPOINT_GLOB = 'api/**/*.{js,ts,tsx}';
+
+function vercelFunctions(): AstroIntegration {
+	let _config: AstroConfig;
+	let output: URL;
+
+	return {
+		name: '@astrojs/vercel',
+		hooks: {
+			'astro:config:setup': ({ config, ignorePages }) => {
+				output = new URL('./.output/', config.projectRoot);
+				config.dist = new URL('./static/', output);
+				config.buildOptions.pageUrlFormat = 'directory';
+				ignorePages(ENDPOINT_GLOB);
+			},
+			'astro:config:done': async ({ config }) => {
+				_config = config;
+			},
+			'astro:build:start': async () => {
+				await fs.rm(output, { recursive: true, force: true });
+			},
+			'astro:build:done': async ({ pages }) => {
+				// Split pages from the rest of files
```

Comment:
If this is only targeting SSG right now, is this "splitting" needed? I thought that Vercel just supported a static output directory (https://docs.astro.build/en/guides/deploy/#vercel)